src/main/java/healthy/lifestyle/backend/admin/users/controller/UserAdminController.java

```diff
@@ -0,0 +1,37 @@
+package healthy.lifestyle.backend.admin.users.controller;
+
+import healthy.lifestyle.backend.admin.users.service.UserAdminService;
+import healthy.lifestyle.backend.users.dto.UserResponseDto;
+import healthy.lifestyle.backend.users.model.Country;
+import healthy.lifestyle.backend.users.model.Role;
+import healthy.lifestyle.backend.validation.*;
+import jakarta.validation.constraints.NotEmpty;
+import java.util.List;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("${api.basePath}/${api.version}/admin")
+public class UserAdminController {
+    private final UserAdminService adminService;
+
+    public UserAdminController(UserAdminService adminService) {
+        this.adminService = adminService;
+    }
+
+    @GetMapping("/users")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    public ResponseEntity<List<UserResponseDto>> getAllUsers(
+            @RequestParam(name = "role", required = true) @NotEmpty Role role,
```

Comment:
I'm not sure that it will convert `?role=ROLE_USER` or `?role=ROLE_ADMIN` to the Role entity automatically. Maybe String role. You can leave it for now, and later see if it works with your HQL query.